extensions/kubernetes-config/deployment/src/main/java/io/quarkus/kubernetes/config/deployment/KubernetesConfigProcessor.java

```diff
@@ -42,8 +42,10 @@ public void handleAccessToSecrets(KubernetesConfigSourceConfig config,
             roleProducer.produce(new KubernetesRoleBuildItem("view-secrets", Collections.singletonList(
                     new KubernetesRoleBuildItem.PolicyRule(
                             Collections.singletonList(""),
+                            null,
+                            config.secrets.orElse(null),
```

Comment:
Looks like CI doesn't like one of the null values.

Reply:
Sorry, I had thought this was a trivial change.  

It's a little confusing because there is usage of config.secrets in warnAboutSecrets - implying that it's supposed to be non-null.  KubernetesTestServerOnProfileTest sees s1, however in the other tests config.secrets is null when this buildstep is run.  It seems related to the config phase.  SecretsEnabled is a build time property, while the secrets are a bootstrap property.  Changing KubernetesConfigSourceConfig to build time does cause the npe to go away.  I'll try to understand a bit better what is going on here.

Reply:
@Ladicek I understand better what is happening / expected.  Either we can't use the secret names to build the policy or the secret names need to be part of the build time config.  The downside of moving them to the build time config would be that it's a breaking change wrt to the policy for anyone who was currently using the env or post build means to supply the secret names.  Do you have a preferred solution here?

Reply:
Ah now I remember! It was exactly the point of "enabling secrets" to allow the list of secrets be runtime-configurable. If the list of secrets was fixed at build time, we wouldn't need a flag to "enable secrets".

Reply:
Updated the pr to just remove list/watch.  However ci is still failing.  On my repo the jdk test failures were due to kafka ssl errors.  Is there anything else I should do for this pr?